Tests/Pcap++Test/Tests/TcpReassemblyTests.cpp

```diff
@@ -1054,3 +1056,68 @@ PTF_TEST_CASE(TestTcpReassemblyMaxSeq)
 	std::string expectedReassemblyData = readFileIntoString(std::string("PcapExamples/one_tcp_stream_output.txt"));
 	PTF_ASSERT_EQUAL(expectedReassemblyData, stats.begin()->second.reassembledData);
 } //TestTcpReassemblyMaxSeq
+
+PTF_TEST_CASE(TestTcpReassemblyDBOOOBufferCC) // TestTcpReassemblyDisableBaseOutOfOrderBufferClearCondition
+{
+	std::string errMsg;
+	std::vector<pcpp::RawPacket> packetStream;
+	TcpReassemblyMultipleConnStats results1;
+	TcpReassemblyMultipleConnStats results2;
+	pcpp::TcpReassemblyConfiguration config1(true, 5, 30,20,true);
+	pcpp::TcpReassemblyConfiguration config2(true, 5, 30,20,false);
+	pcpp::TcpReassembly tcpReassembly1(tcpReassemblyMsgReadyCallback, &results1, tcpReassemblyConnectionStartCallback, tcpReassemblyConnectionEndCallback, config1);
+	pcpp::TcpReassembly tcpReassembly2(tcpReassemblyMsgReadyCallback, &results2, tcpReassemblyConnectionStartCallback, tcpReassemblyConnectionEndCallback, config2);
+	PTF_ASSERT_TRUE(readPcapIntoPacketVec("PcapExamples/one_tcp_stream.pcap", packetStream, errMsg));
+
+	// unserting a data packet from reverse direction b/w swap 2 consequent data packets
+	std::swap(packetStream[12], packetStream[13]);
+	std::swap(packetStream[13], packetStream[18]);
+
+	TcpReassemblyMultipleConnStats tcpReassemblyResults;
+	for (std::vector<pcpp::RawPacket>::iterator iter = packetStream.begin(); iter != packetStream.end(); iter++)
+	{
+		pcpp::Packet packet(&(*iter));
+		tcpReassembly1.reassemblePacket(packet);
+		tcpReassembly2.reassemblePacket(packet);
+	}
+
+	tcpReassembly1.closeAllConnections();
+	tcpReassembly2.closeAllConnections();
+
+	TcpReassemblyMultipleConnStats::Stats &stats1 = results1.stats;
+	TcpReassemblyMultipleConnStats::Stats &stats2 = results2.stats;
+	PTF_ASSERT_EQUAL(stats1.size(), 1);
+	PTF_ASSERT_EQUAL(stats2.size(), 1);
+	PTF_ASSERT_EQUAL(stats1.begin()->second.numOfDataPackets, 18);
+	PTF_ASSERT_EQUAL(stats2.begin()->second.numOfDataPackets, 19);
+
+	packetStream.clear();
+	tcpReassemblyResults.clear();
+} // TestTcpReassemblyDBOOOBufferCC
+
+PTF_TEST_CASE(TestTcpReassemblyTstamps) // TestTcpReassemblyTimeStamps
```

Comment:
If we change the test name I don't think we need the `// TestTcpReassemblyTimeStamps` comment